plugins/opensslcryptoplugin/evp_p.h

```diff
@@ -232,11 +232,57 @@ int osslevp_aes_decrypt_ciphertext(const unsigned char *init_vector,
     *decrypted = plaintext;
 
     /* Clean up the decryption context */
-    EVP_CIPHER_CTX_cleanup(&decryption_context);
+    EVP_CIPHER_CTX_free(decryption_context);
     plaintext_length = update_length + final_length;
     return plaintext_length;
 }
 
+/*
+    Implements signing according to:
+    https://wiki.openssl.org/index.php/EVP_Signing_and_Verifying
```

Comment:
Add a comment to document the ownership semantics of the signature outparam, i.e. that the caller must free with OPENSSL_free()

Reply:
Ok!